src/main/scala/it/unibo/authsim/library/user/builder/AutoUserBuilder.scala

```diff
@@ -0,0 +1,27 @@
+package it.unibo.authsim.library.user.builder
+
+import it.unibo.authsim.library.dsl.policy.checkers.StringPolicyChecker
+import it.unibo.authsim.library.dsl.policy.model.StringPolicies.{PasswordPolicy, UserIDPolicy}
+import it.unibo.authsim.library.user.model.User
+//Non sono sicurissima che qui sia sensata una classe o se sarebbe meglio un object
```

Comment:
If the builder can be configured in more than 1 way, than a `class` has its place; if the builder has no configuration you can make it an `object`.

Reply:
Alex what do you mean with _in more than one way_?

Reply:
If you foresee a moment when you will have to call `new UserAutoBuilder(XYZ)` or `new UserAutoBuilder(ABC)` which leads to a different behaviour then you have more than 1 possible configuration.

Another case is when you want more than one instance (you call more than 1 time `new`) at a given time, `UserAutoBuilder` should be a class in this case too.

Reply:
I think that having a different kinds of users is a possibility that could happen in the future, do you agree?

Reply:
Yes, also, one could need to build milions of users concurrently, so they would need more than one builder.

For both reasons, I'd keep it a class.